src/utility-types.ts

```diff
@@ -7,23 +7,16 @@ import type { KillBehavior } from './worker/worker-options'
  */
 export type Draft<T> = { -readonly [P in keyof T]?: T[P] }
 
-/**
- * Serializable primitive JSON value.
- */
-export type JSONPrimitive = number | boolean | string | null
 /**
  * Serializable JSON value.
  */
-// eslint-disable-next-line no-use-before-define
-export type JSONValue = JSONPrimitive | JSONArray | JSONObject
-/**
- * Serializable JSON object.
- */
-export type JSONObject = { [k: string]: JSONValue }
-/**
- * Serializable JSON array.
- */
-export type JSONArray = Array<JSONValue>
+export type JSONValue =
+  | string
+  | number
+  | boolean
+  | null
+  | JSONValue[]
+  | Record<string, JSONValue>
```

Comment:
So this fix should work for now, until Microsoft has fixed it

```suggestion
  | { [k: string]: JSONValue }
```

Reply:
With that type definition, you get a missing Index compilation error with a 
WorkerData custom interface like `{ index: number, data: any }` which is perfectly legal. 
With the Record type, it just works as expected. 
I do see the point of not using the correct, bug free, upstream maintained and working type definition shortcuts because of bugs that does not belong to poolifier. 
The code change is that PR made poolifier 2.0.0 just usable, without it's not usable.

Reply:
THAT one is the real bug in our tool :eyes: 
damnd!

Okay so @pioardi are unintentionally right and we need to use unknwon everywhere + use jsdoc

Reply:
@jerome-benoit I will make the change in this branch, so please give me a minute or two

Reply:
Well, I prefer to have type for workerData that works than any.

Reply:
Defining `WorkerData` on your own is exactly possible and then should be used as the first generic parameter for the Worker and the Pool

Reply:
So like this: 
![image](https://user-images.githubusercontent.com/7195563/108533112-e5de8500-72d8-11eb-94a4-bdaca473f7eb.png)

Reply:
Fair enough. Let the user enforce its in/out data types.